### ISSUE TITLE
Incorrect Datepicker rendering on IE11 and friends (IE11+)

### ISSUE THREAD

**Issue by Inlesco** [zippyui/react-date-picker#166]
The Datepicker's calendar renders incorrectly on IE11 and friends.

If you take a look at the picture below, it seems that the Calendar's width is set to input's width in IE11 and it never goes past that.
I've changed the width of calendar's parent and it really seems that on IE the calendar has width of 100% inherited from its parent. How come?

Is this because:

- you use react-flex and it renders problematically on IE11+?
- calculations for Calendar's width (positioning relative to input and actual width) are off (only on IE11?)?

Issue image:
![screenshot_30](https://cloud.githubusercontent.com/assets/5101235/18513361/dac7b258-7a95-11e6-930f-af082f31c5e6.png)

**Comment by MrSauceman** [comment]
I had this same issue. Finally figured it out.  I added this css and it fixes the IE clipping issue:

`.react-date-field__picker {
    display: block; // ie fix
}`

I added one other css fix for ie, as well.  This one fixes the day on the calendar not being in the center of its circle:

`.dp-week {
    line-height: 31px; // ie fix
}`

**Comment by Inlesco** [comment]
@MrSauceman Yeah, I already solved all of this by doing exactly the same. Thanks for the input!

Hopefully, the devs will fix such small quirks.

**Comment by algreasley** [comment]
That's a great help, thanks!

I also needed to hide the IE clear button on the input as this clashed with RDP's clear button:

    // http://stackoverflow.com/questions/20473413/remove-ie11s-clear-field-x-button-on-windows-8
    .react-date-field__input::-ms-clear {
        display: none;
        width:0;
        height:0;
    }

BTW, website failure seems to be failure to load "react-date-picker/input.css"

**Comment by utsavig** [comment]
How do you fix the month and year viewer in IE 11?  
.react-date-field__picker { display: block; // ie fix }   fixes the display for the picker but not for the month and year viewer.

![image](https://cloud.githubusercontent.com/assets/11548336/20911164/e6ead058-bbb1-11e6-80aa-ee8591ae2462.png)

**Comment by utsavig** [comment]
also, can you hide months from the month and year viewer?

**Comment by algreasley** [comment]
I ended up hiding the whole popup, as it is not critical to the operation of the picker, and it does not look good on IE.

The magic property is 'enableHistoryView' on the Calendar (aka DatePicker)

**Comment by utsavig** [comment]
Thank you @algreasley.  Year Viewer is handy for things like Date Of Birth where User might have to scroll back a few years.

**Comment by rgabs** [comment]
@utsavig The iE issue is fixed in this PR: https://github.com/zippyui/react-date-picker/pull/240

**Comment by utsavig** [comment]
Thanks @rgabs !!
